packages/create-turbo/src/commands/create/createProject.ts

```diff
@@ -0,0 +1,231 @@
+      );
+      process.exit(1);
+    }
+  } else {
+    const found = await existsInRepo(example);
+
+    if (!found) {
+      error(
+        `Could not locate an example named ${chalk.red(
+          `"${example}"`
+        )}. It could be due to the following:\n`,
+        `1. Your spelling of example ${chalk.red(
+          `"${example}"`
+        )} might be incorrect.\n`,
+        `2. You might not be connected to the internet or you are behind a proxy.`
+      );
+      process.exit(1);
+    }
+  }
+
+  const root = path.resolve(appPath);
+
+  if (!(await isWriteable(path.dirname(root)))) {
+    error(
+      "The application path is not writable, please check folder permissions and try again."
+    );
+    error("It is likely you do not have write permissions for this folder.");
+    process.exit(1);
+  }
+
+  const appName = path.basename(root);
+  try {
+    await fs.mkdir(root, { recursive: true });
+  } catch (err) {
+    error("Unable to create project directory");
+    console.error(err);
+    process.exit(1);
+  }
+  const { isEmpty, conflicts } = isFolderEmpty(root);
+  if (!isEmpty) {
+    error(
+      `${chalk.dim(root)} has ${conflicts.length} conflicting ${
+        conflicts.length === 1 ? "file" : "files"
+      } - please try a different location`
+    );
+    process.exit(1);
+  }
+
+  const originalDirectory = process.cwd();
+  process.chdir(root);
+
+  /**
+   * clone the example repository
+   */
+  const loader = turboLoader("Downloading files...");
+  try {
+    if (repoInfo) {
+      const repoInfo2 = repoInfo;
+      console.log(
+        `\nDownloading files from repo ${chalk.cyan(
+          example
+        )}. This might take a moment.`
+      );
+      console.log();
+      loader.start();
+      await retry(() => downloadAndExtractRepo(root, repoInfo2), {
+        retries: 3,
+      });
+    } else {
+      console.log(
+        `\nDownloading files${
+          !isDefaultExample ? ` for example ${chalk.cyan(example)}` : ""
+        }. This might take a moment.`
+      );
+      console.log();
+      loader.start();
+      await retry(() => downloadAndExtractExample(root, example), {
+        retries: 3,
+      });
+    }
+  } catch (reason) {
+    function isErrorLike(err: unknown): err is { message: string } {
+      return (
+        typeof err === "object" &&
+        err !== null &&
+        typeof (err as { message?: unknown }).message === "string"
+      );
+    }
+    throw new DownloadError(isErrorLike(reason) ? reason.message : reason + "");
+  } finally {
+    loader.stop();
+  }
+
+  const rootPackageJsonPath = path.join(root, "package.json");
+  const rootMetaJsonPath = path.join(root, "meta.json");
+  const ignorePath = path.join(root, ".gitignore");
+  const hasPackageJson = fs.existsSync(rootPackageJsonPath);
+  const isTurboExample =
+    repoInfo && repoInfo.username === "vercel" && repoInfo.name === "turbo";
+  const availableScripts = [];
+
+  // remove the meta file from turbo examples
```

Comment:
can we model this as a "transformer" that gets applied to known repos? it will be easier to rip it out if we decide we don't want our examples to be fancy and have "compile" time logic to build.

Reply:
We could, but I'm not sure if that's worth the time right now when we would only have one transformer. I'm usually a fan of optimizing later when the patterns emerge. This is a pretty straight-forward case. The example is either one of ours (in which case we have a few special rules) or it's not (we don't do anything)